### ISSUE TITLE
Normalize layer equals to L2Normalization+Scale Layer?

### ISSUE THREAD

**Issue by zhreshold** [weiliu89/caffe#202]
First thanks for the brilliant work, I've spent some time reading and rewriting the code you provided.
The reimplementation on MXNet  is almost finished. https://github.com/zhreshold/mxnet-ssd

Unfortunately, I still cannot train a model which is as good as the one you provided. The best I can get right now is only about 55% mAP on VOC2007 test set. (I did not continue training since validation status stuck for quite a while, at about 300 epoch, it just don't converge as fast as yours, regardless of different learning rates I tried).

I think I have implemented almost everything you have mentioned in paper or what's provided in code (data augmentation, hard negative mining...). The NMS is slightly different but I don't think that would make too much problem during evaluation.

For now, the biggest question is the Normalize layer you added to relu4_3 layer.
* It is equals to L2Normalization to each channel and add a scaling layer to spatial element?
* Why the normalize is only applied to relu4_3? I can see the relu7 output also has large magnitude. And why not apply to all from_layers?

I'd appreciate if you got some time to answer my questions.

**Comment by zhreshold** [comment]
Problem solved, now producing same results.
Also the model can be converted from caffe to mxnet now.

**Comment by KeyKy** [comment]
@zhreshold could you share how do you solve your problem and the key points of the problem? thanks.
